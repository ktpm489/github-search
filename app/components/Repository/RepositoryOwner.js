import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, StyleSheet } from 'react-native';
import { Avatar } from 'react-native-elements';

const RepositoryOwner = ({
  avatarUrl,
  name,
  login,
  height,
  width,
  containerStyle,
  flexDirection,
  infoContainerStyle,
  onAvatarPress,
}) => (
  <View style={[styles.container, { flexDirection }]}>
    <Avatar
      height={height}
      width={width}
      rounded
      source={{ uri: avatarUrl }}
      containerStyle={containerStyle}
      onPress={onAvatarPress}
    />
    <View style={infoContainerStyle}>
      <Text style={styles.name}>{name}</Text>
      <Text style={styles.login}>{login}</Text>
    </View>
  </View>
);

RepositoryOwner.propTypes = {
  avatarUrl: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  login: PropTypes.string.isRequired,
  height: PropTypes.number,
  width: PropTypes.number,
  containerStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  flexDirection: PropTypes.oneOf(['row', 'column']),
  infoContainerStyle: PropTypes.oneOfType([PropTypes.number, PropTypes.object]),
  onAvatarPress: PropTypes.func.isRequired,
};

RepositoryOwner.defaultProps = {
  containerStyle: {},
  height: 100,
  width: 100,
  flexDirection: 'column',
  infoContainerStyle: {},
};

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    alignItems: 'center',
    marginBottom: 10,
  },
  infoContainer: {
    display: 'flex',
    flex: 1,
    marginLeft: 10,
    justifyContent: 'center',
  },
  name: {
    color: '#000',
    fontSize: 22,
    fontWeight: '700',
  },
  login: {
    color: '#6C7680',
    fontSize: 18,
  },
});

export default RepositoryOwner;
