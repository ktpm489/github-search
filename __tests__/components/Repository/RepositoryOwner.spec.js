import React from 'react';
import { shallow } from 'enzyme';

import RepositoryOwner from '../../../app/components/Repository/RepositoryOwner';

describe('<RepositoryOwner />', () => {
  it('renders correctly with all required props', () => {
    const wrapper = shallow(
      <RepositoryOwner
        name="Parker Ziegler"
        avatarUrl="mock"
        login="parkerziegler"
        onAvatarPress={jest.fn()}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly with optional props', () => {
    const wrapper = shallow(
      <RepositoryOwner
        name="Parker Ziegler"
        avatarUrl="mock"
        login="parkerziegler"
        onAvatarPress={jest.fn()}
        height={60}
        width={60}
        containerStyle={{
          margin: '0 auto',
        }}
        flexDirection="row"
        infoContainerStyle={{
          alignItems: 'flex-end',
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
