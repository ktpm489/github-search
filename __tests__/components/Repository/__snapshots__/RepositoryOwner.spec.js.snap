// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepositoryOwner /> renders correctly with all required props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryOwner
    avatarUrl="mock"
    containerStyle={Object {}}
    flexDirection="column"
    height={100}
    infoContainerStyle={Object {}}
    login="parkerziegler"
    name="Parker Ziegler"
    onAvatarPress={[MockFunction]}
    width={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Avatar
          containerStyle={Object {}}
          editButton={
            Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            }
          }
          height={100}
          onEditPress={null}
          onPress={[MockFunction]}
          rounded={true}
          showEditButton={false}
          source={
            Object {
              "uri": "mock",
            }
          }
          width={100}
        />,
        <View
          style={Object {}}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontSize": 22,
                "fontWeight": "700",
              }
            }
          >
            Parker Ziegler
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6C7680",
                "fontSize": 18,
              }
            }
          >
            parkerziegler
          </Text>
        </View>,
      ],
      "style": Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": 10,
        },
        Object {
          "flexDirection": "column",
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {},
          "editButton": Object {
            "iconColor": "#fff",
            "iconName": "mode-edit",
            "iconType": "material",
            "size": null,
            "style": null,
            "underlayColor": "#000",
          },
          "height": 100,
          "onEditPress": null,
          "onPress": [MockFunction],
          "rounded": true,
          "showEditButton": false,
          "source": Object {
            "uri": "mock",
          },
          "width": 100,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                }
              }
            >
              Parker Ziegler
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                }
              }
            >
              parkerziegler
            </Text>,
          ],
          "style": Object {},
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Parker Ziegler",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#000",
                "fontSize": 22,
                "fontWeight": "700",
              },
            },
            "ref": null,
            "rendered": "Parker Ziegler",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "parkerziegler",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#6C7680",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "parkerziegler",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Avatar
            containerStyle={Object {}}
            editButton={
              Object {
                "iconColor": "#fff",
                "iconName": "mode-edit",
                "iconType": "material",
                "size": null,
                "style": null,
                "underlayColor": "#000",
              }
            }
            height={100}
            onEditPress={null}
            onPress={[MockFunction]}
            rounded={true}
            showEditButton={false}
            source={
              Object {
                "uri": "mock",
              }
            }
            width={100}
          />,
          <View
            style={Object {}}
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                }
              }
            >
              Parker Ziegler
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                }
              }
            >
              parkerziegler
            </Text>
          </View>,
        ],
        "style": Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "marginBottom": 10,
          },
          Object {
            "flexDirection": "column",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {},
            "editButton": Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            },
            "height": 100,
            "onEditPress": null,
            "onPress": [MockFunction],
            "rounded": true,
            "showEditButton": false,
            "source": Object {
              "uri": "mock",
            },
            "width": 100,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 22,
                    "fontWeight": "700",
                  }
                }
              >
                Parker Ziegler
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#6C7680",
                    "fontSize": 18,
                  }
                }
              >
                parkerziegler
              </Text>,
            ],
            "style": Object {},
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Parker Ziegler",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                },
              },
              "ref": null,
              "rendered": "Parker Ziegler",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "parkerziegler",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                },
              },
              "ref": null,
              "rendered": "parkerziegler",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<RepositoryOwner /> renders correctly with optional props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryOwner
    avatarUrl="mock"
    containerStyle={
      Object {
        "margin": "0 auto",
      }
    }
    flexDirection="row"
    height={60}
    infoContainerStyle={
      Object {
        "alignItems": "flex-end",
      }
    }
    login="parkerziegler"
    name="Parker Ziegler"
    onAvatarPress={[MockFunction]}
    width={60}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Avatar
          containerStyle={
            Object {
              "margin": "0 auto",
            }
          }
          editButton={
            Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            }
          }
          height={60}
          onEditPress={null}
          onPress={[MockFunction]}
          rounded={true}
          showEditButton={false}
          source={
            Object {
              "uri": "mock",
            }
          }
          width={60}
        />,
        <View
          style={
            Object {
              "alignItems": "flex-end",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#000",
                "fontSize": 22,
                "fontWeight": "700",
              }
            }
          >
            Parker Ziegler
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#6C7680",
                "fontSize": 18,
              }
            }
          >
            parkerziegler
          </Text>
        </View>,
      ],
      "style": Array [
        Object {
          "alignItems": "center",
          "display": "flex",
          "marginBottom": 10,
        },
        Object {
          "flexDirection": "row",
        },
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "containerStyle": Object {
            "margin": "0 auto",
          },
          "editButton": Object {
            "iconColor": "#fff",
            "iconName": "mode-edit",
            "iconType": "material",
            "size": null,
            "style": null,
            "underlayColor": "#000",
          },
          "height": 60,
          "onEditPress": null,
          "onPress": [MockFunction],
          "rounded": true,
          "showEditButton": false,
          "source": Object {
            "uri": "mock",
          },
          "width": 60,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                }
              }
            >
              Parker Ziegler
            </Text>,
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                }
              }
            >
              parkerziegler
            </Text>,
          ],
          "style": Object {
            "alignItems": "flex-end",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "Parker Ziegler",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#000",
                "fontSize": 22,
                "fontWeight": "700",
              },
            },
            "ref": null,
            "rendered": "Parker Ziegler",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "parkerziegler",
              "ellipsizeMode": "tail",
              "style": Object {
                "color": "#6C7680",
                "fontSize": 18,
              },
            },
            "ref": null,
            "rendered": "parkerziegler",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Avatar
            containerStyle={
              Object {
                "margin": "0 auto",
              }
            }
            editButton={
              Object {
                "iconColor": "#fff",
                "iconName": "mode-edit",
                "iconType": "material",
                "size": null,
                "style": null,
                "underlayColor": "#000",
              }
            }
            height={60}
            onEditPress={null}
            onPress={[MockFunction]}
            rounded={true}
            showEditButton={false}
            source={
              Object {
                "uri": "mock",
              }
            }
            width={60}
          />,
          <View
            style={
              Object {
                "alignItems": "flex-end",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                }
              }
            >
              Parker Ziegler
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                }
              }
            >
              parkerziegler
            </Text>
          </View>,
        ],
        "style": Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "marginBottom": 10,
          },
          Object {
            "flexDirection": "row",
          },
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "containerStyle": Object {
              "margin": "0 auto",
            },
            "editButton": Object {
              "iconColor": "#fff",
              "iconName": "mode-edit",
              "iconType": "material",
              "size": null,
              "style": null,
              "underlayColor": "#000",
            },
            "height": 60,
            "onEditPress": null,
            "onPress": [MockFunction],
            "rounded": true,
            "showEditButton": false,
            "source": Object {
              "uri": "mock",
            },
            "width": 60,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#000",
                    "fontSize": 22,
                    "fontWeight": "700",
                  }
                }
              >
                Parker Ziegler
              </Text>,
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#6C7680",
                    "fontSize": 18,
                  }
                }
              >
                parkerziegler
              </Text>,
            ],
            "style": Object {
              "alignItems": "flex-end",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "Parker Ziegler",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#000",
                  "fontSize": 22,
                  "fontWeight": "700",
                },
              },
              "ref": null,
              "rendered": "Parker Ziegler",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "parkerziegler",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": "#6C7680",
                  "fontSize": 18,
                },
              },
              "ref": null,
              "rendered": "parkerziegler",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
